diff --git a/packages/flutter_test/lib/src/binding.dart b/packages/flutter_test/lib/src/binding.dart
index 7d450be42..e158e33af 100644
--- a/packages/flutter_test/lib/src/binding.dart
+++ b/packages/flutter_test/lib/src/binding.dart
@@ -1763,9 +1763,10 @@ class _MockHttpResponse extends Stream<List<int>> implements HttpClientResponse
   @override
   int get contentLength => -1;
 
-  // @override
-  // TODO(tvolkert): Uncomment @override annotation once SDK change lands.
-  bool get autoUncompress => true;
+  @override
+  HttpClientResponseCompressionState get compressionState {
+    return HttpClientResponseCompressionState.decompressed;
+  }
 
   @override
   List<Cookie> get cookies => null;
diff --git a/packages/flutter_tools/lib/src/base/io.dart b/packages/flutter_tools/lib/src/base/io.dart
index 4c9d2250b..8b3535d54 100644
--- a/packages/flutter_tools/lib/src/base/io.dart
+++ b/packages/flutter_tools/lib/src/base/io.dart
@@ -47,6 +47,7 @@ export 'dart:io'
         HttpClient,
         HttpClientRequest,
         HttpClientResponse,
+        HttpClientResponseCompressionState,
         HttpHeaders,
         HttpRequest,
         HttpServer,
diff --git a/packages/flutter_tools/test/commands/create_test.dart b/packages/flutter_tools/test/commands/create_test.dart
index b5453cbb1..f7cdf6164 100644
--- a/packages/flutter_tools/test/commands/create_test.dart
+++ b/packages/flutter_tools/test/commands/create_test.dart
@@ -1172,6 +1172,11 @@ class MockHttpClientResponse extends Stream<List<int>> implements HttpClientResp
   @override
   String get reasonPhrase => '<reason phrase>';
 
+  @override
+  HttpClientResponseCompressionState get compressionState {
+    return HttpClientResponseCompressionState.decompressed;
+  }
+
   @override
   StreamSubscription<List<int>> listen(
     void onData(List<int> event), {
