diff --git a/DEPS b/DEPS
index f2b3cc9..88557e8 100644
--- a/DEPS
+++ b/DEPS
@@ -72,6 +72,6 @@ vars = {
   'dart_oauth2_tag': '1.2.1',
   'dart_observatory_pub_packages_rev': '0894122173b0f98eb08863a7712e78407d4477bc',
-  'dart_package_config_tag': '2453cd2e78c2db56ee2669ced17ce70dd00bf576',
+  'dart_package_config_tag': '3401e2897b3cf46e64966e2ba19ed7032501cf41',
   'dart_package_resolver_tag': '1.0.10',
   'dart_path_tag': '1.6.2',
   'dart_pedantic_tag': 'v1.8.0',
 
