diff --git a/BUILD.gn b/BUILD.gn
index 2766d1b98..828196f8d 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -60,8 +60,8 @@ group("flutter") {
 
     if (is_linux) {
       public_deps += [
-        "$flutter_root/shell/platform/common/cpp/client_wrapper:client_wrapper_unittests",
-        "$flutter_root/shell/platform/glfw/client_wrapper:client_wrapper_glfw_unittests",
+#        "$flutter_root/shell/platform/common/cpp/client_wrapper:client_wrapper_unittests",
+#        "$flutter_root/shell/platform/glfw/client_wrapper:client_wrapper_glfw_unittests",
       ]
     }
   }
