diff --git a/DEPS b/DEPS
index c395874b2..36f4de34b 100644
--- a/DEPS
+++ b/DEPS
@@ -96,6 +96,7 @@ vars = {
   'dart_term_glyph_tag': '1.0.1',
   'dart_test_reflective_loader_tag': '0.1.8',
   'dart_test_tag': 'test-v1.6.4',
+  'dart_tflite_native_rev': '712b8a93fbb4caf83ffed37f154da88c2a517a91',
   'dart_typed_data_tag': '1.1.6',
   'dart_usage_tag': '3.4.0',
   'dart_watcher_rev': '0.9.7+12-pub',
@@ -343,6 +344,9 @@ deps = {
   'src/third_party/dart/third_party/pkg/test':
    Var('dart_git') + '/test.git' + '@' + Var('dart_test_tag'),
 
+  'src/third_party/dart/third_party/pkg/tflite_native':
+   Var('dart_git') + '/tflite_native.git' + '@' + Var('dart_tflite_native_rev'),
+
   'src/third_party/dart/third_party/pkg/test_reflective_loader':
    Var('dart_git') + '/test_reflective_loader.git' + '@' + Var('dart_test_reflective_loader_tag'),
 
@@ -491,6 +495,16 @@ deps = {
      'dep_type': 'cipd',
    },
 
+   'src/third_party/dart/pkg/analysis_server/language_model': {
+     'packages': [
+       {
+        'package': 'dart/language_model',
+        'version': 'gABkW8D_-f45it57vQ_ZTKFwev16RcCjvrdTCytEnQgC',
+       }
+     ],
+     'dep_type': 'cipd',
+   },
+
   'src/flutter/third_party/gn': {
     'packages': [
       {
