diff --git a/frontend_server/lib/server.dart b/frontend_server/lib/server.dart
index 804c5699e..572087e9d 100644
--- a/frontend_server/lib/server.dart
+++ b/frontend_server/lib/server.dart
@@ -39,6 +39,11 @@ class _FlutterFrontendCompiler implements frontend.CompilerInterface{
     _compiler.acceptLastDelta();
   }
 
+  @override
+  Future<Null> rejectLastDelta() async {
+    return _compiler.rejectLastDelta();
+  }
+
   @override
   void invalidate(Uri uri) {
     _compiler.invalidate(uri);
