diff --git a/packages/flutter/test/material/theme_test.dart b/packages/flutter/test/material/theme_test.dart
index 52b033ed829..6dc78024951 100644
--- a/packages/flutter/test/material/theme_test.dart
+++ b/packages/flutter/test/material/theme_test.dart
@@ -459,6 +459,10 @@ class _TextStyleProxy implements TextStyle {
   @override TextBaseline get textBaseline => _delegate.textBaseline;
   @override double get wordSpacing => _delegate.wordSpacing;
 
+  @override
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.debug}) =>
+      super.toString();
+
   @override
   DiagnosticsNode toDiagnosticsNode({String name, DiagnosticsTreeStyle style}) {
     throw new UnimplementedError();
diff --git a/packages/flutter/test/rendering/proxy_box_test.dart b/packages/flutter/test/rendering/proxy_box_test.dart
index f036704de8f..6ba13079b62 100644
--- a/packages/flutter/test/rendering/proxy_box_test.dart
+++ b/packages/flutter/test/rendering/proxy_box_test.dart
@@ -292,4 +292,7 @@ class _FakeTicker implements Ticker {
 
   @override
   void unscheduleTick() {}
+
+  @override
+  String toString({bool debugIncludeStack = false}) => super.toString();
 }
