diff --git a/frontend_server/lib/server.dart b/frontend_server/lib/server.dart
index 44a3d69b9..9620a5de0 100644
--- a/frontend_server/lib/server.dart
+++ b/frontend_server/lib/server.dart
@@ -85,11 +85,6 @@ Future<int> starter(
       StringSink output,
     }) async {
   ArgResults options;
-  frontend.argParser
-    ..addFlag('track-widget-creation',
-      help: 'Run a kernel transformer to track creation locations for widgets.',
-      defaultsTo: false);
-
   try {
     options = frontend.argParser.parse(args);
   } catch (error) {
