diff --git a/DEPS b/DEPS
index 42d057195..716e4b181 100644
--- a/DEPS
+++ b/DEPS
@@ -53,7 +53,7 @@ vars = {
   'dart_dart2js_info_tag': '0.6.0',
   'dart_dart_style_tag': '1.3.2',
   'dart_ffi_tag': 'ea88d71b043ee14b268c3aedff14e9eb32e20959',
-  'dart_fixnum_tag': '0.10.9',
+  'dart_fixnum_tag': '9b38f49f6679654d66a363e69e48173cca07e882',
   'dart_glob_tag': '1.1.7',
   'dart_html_tag': '0.14.0+1',
   'dart_http_multi_server_tag': '2.0.5',
