diff --git a/DEPS b/DEPS
index 17f3f9f68..39b78c735 100644
--- a/DEPS
+++ b/DEPS
@@ -198,6 +198,9 @@ deps = {
   'src/third_party/dart/third_party/pkg/ffi':
    Var('dart_git') + '/ffi.git@a5d4232cd38562c75a3ed847baa340e399538028',
 
+  'src/third_party/dart/third_party/pkg/file':
+   Var('dart_git') + '/external/github.com/google/file.dart@0e09370f581ab6388d46fda4cdab66638c0171a1',
+
   'src/third_party/dart/third_party/pkg/fixnum':
    Var('dart_git') + '/fixnum.git@16d3890c6dc82ca629659da1934e412292508bba',
 
