diff --git a/DEPS b/DEPS
index 2f5a8532b..c4cb5da32 100644
--- a/DEPS
+++ b/DEPS
@@ -47,7 +47,7 @@ vars = {
   'dart_convert_tag': '2.0.2',
   'dart_crypto_tag': '2.0.6',
   'dart_csslib_tag': '0.14.4+1',
-  'dart_dart2js_info_tag': '0.5.13',
+  'dart_dart2js_info_tag': '0.5.15',
   'dart_dart_style_tag': '1.2.0',
   'dart_dartdoc_tag': 'v0.24.1',
   'dart_fixnum_tag': '0.10.8',
@@ -60,7 +60,7 @@ vars = {
   'dart_http_throttle_tag': '1.0.2',
   'dart_intl_tag': '0.15.7',
   'dart_json_rpc_2_tag': '2.0.9',
-  'dart_linter_tag': '0.1.71',
+  'dart_linter_tag': '0.1.75',
   'dart_logging_tag': '0.11.3+2',
   'dart_markdown_tag': '2.0.2',
   'dart_matcher_tag': '0.12.3',
diff --git a/lib/snapshot/BUILD.gn b/lib/snapshot/BUILD.gn
index ef06063a2..4db8600ea 100644
--- a/lib/snapshot/BUILD.gn
+++ b/lib/snapshot/BUILD.gn
@@ -53,9 +53,6 @@ compiled_action("generate_snapshot_bin") {
   ]
 
   args = [
-    "--strong",
-    "--sync-async",
-    "--reify-generic-functions",
     "--snapshot_kind=core",
     "--enable_mirrors=false",
     "--vm_snapshot_data=" + rebase_path(vm_snapshot_data),
